src/Hangfire.MemoryStorage/Utilities/StateHistoryList.cs

```diff
@@ -0,0 +1,83 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Hangfire.Storage.Monitoring;
+
+namespace Hangfire.MemoryStorage.Utilities
+{
+    public class StateHistoryList : IList<StateHistoryDto>
+    {
+        private readonly IList<StateHistoryDto> _list = new List<StateHistoryDto>();
+
+        public IEnumerator<StateHistoryDto> GetEnumerator()
+        {
+            return _list.GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        public void Add(StateHistoryDto item)
+        {
+            if (_list.Any(t => t.StateName == "Created"))

```

Comment:
Custom collection type to guard against multiple 'Created' history entries from being added.